src/Command/Util/ApigeeEdgeManagementCliService.php

```diff
@@ -0,0 +1,330 @@
+        // Role does not exist.
+        return FALSE;
+      }
+      // Any other response was an exception.
+      throw $exception;
+    }
+    // If 200 response, role exists.
+    return TRUE;
+  }
+
+  /**
+   * Validate the Apigee Edge org connection settings.
+   *
+   * @param \Symfony\Component\Console\Style\StyleInterface $io
+   *   The IO interface of the CLI tool calling the method.
+   * @param callable $t
+   *   The translation function akin to t().
+   * @param string $org
+   *   The Edge org to connect to.
+   * @param string $email
+   *   The email of an Edge user with org admin role to make Edge API calls.
+   * @param string $password
+   *   The password of an Edge user email to make Edge API calls.
+   * @param string $base_url
+   *   The base url of the Edge API.
+   *
+   * @return bool
+   *   Return true if the Edge API can be called, false if it cannot.
+   */
+  public function isValidEdgeCredentials(StyleInterface $io, callable $t, string $org, string $email, string $password, string $base_url) {
+    $url = $base_url . '/o/' . $org;
+    try {
+      $response = $this->httpClient->get($url, [
+        'auth' => [$email, $password],
+        'headers' => ['Accept' => 'application/json'],
+      ]);
+    }
+    catch (TransferException $exception) {
+      $this->handleHttpClientExceptions($exception, $io, $t, $url, $org, $email);
+      return FALSE;
+    }
+
+    $body = json_decode($response->getBody());
+    if (JSON_ERROR_NONE !== json_last_error()) {
+      $io->error($t('Unable to parse response from Apigee Edge into JSON. !error ', ['error' => json_last_error_msg()]));
+      $io->section($t('Server response from :url', [':url' => $url]));
+      $io->text($response->getBody());
+      return FALSE;
+    }
+    if (!isset($body->name)) {
+      $io->warning($t('The response from :url did not contain the org name.', [':url' => $url]));
+    }
+    else {
+      $io->success($t('Connected to Edge org :org.', [':org' => $body->name]));
+    }
+    return TRUE;
+  }
+
+  /**
+   * Print out helpful information to user running command when error happens.
+   *
+   * @param \GuzzleHttp\Exception\TransferException $exception
+   *   The exception thrown.
+   * @param \Symfony\Component\Console\Style\StyleInterface $io
+   *   The IO interface of the CLI tool calling the method.
+   * @param callable $t
+   *   The translation function akin to t().
+   * @param string $url
+   *   The url being connected to.
+   * @param string $org
+   *   The organization to connect to.
+   * @param string $email
+   *   The email of an Edge user with org admin role to make Edge API calls.
+   */
+  public function handleHttpClientExceptions(TransferException $exception, StyleInterface $io, callable $t, string $url, string $org, string $email): void {
+    $io->error($t('Error connecting to Apigee Edge. :exception_message', [':exception_message' => $exception->getMessage()]));
+    switch ($exception->getCode()) {
```

Comment:
add a default case to handle when the error is not 0,401,403,302?

Reply:
Line 304 `$io->error()` will display out info about any error, the switch statement is only to add extra info for the end user for common use cases:

<img width="948" alt="Screen Shot 2019-10-04 at 3 53 57 PM" src="https://user-images.githubusercontent.com/613093/66244904-87067480-e6bf-11e9-8298-14dc9726462c.png">